### ISSUE TITLE
threading warning when open UIImagePicker

### ISSUE THREAD

**Issue by MoElnaggar14** [helpscout/beacon-ios-sdk#46]
Thank you for taking the time to submit an issue with all the details shown below.  Our engineering team monitors issues submitted and strives to respond with 1-2 business days.

* Beacon install method: [ ] Carthage, [x] CocoaPods, [ ] Manual

**Screenshots or GIFs**
![Screen Shot 2019-11-18 at 11 35 51 AM](https://user-images.githubusercontent.com/21294763/69041270-e6161380-09f7-11ea-95a1-b97f03faaa8d.png)

**Comment by codenameDuchess** [comment]
Hi @MoElnaggar14 

Thanks for submitting this issue. Would you please double check that when you call HSBeacon.open you're doing so on your application's main thread?  

We are unable to recreate this issue locally so if you still see the issue after that double-check please provide us with a code snippet of how you are opening the Beacon so we can investigate further. 

(If you don't feel comfortable posting your code here please send us a note at help@helpscout.com).

**Comment by imanoupetit** [comment]
I could reproduce it with the BeaconExample app (version 1.0 build 35):
- Tap **Take photo**
- Tap **OK** when **"BeaconExample" Would Like to Access Your Photos** alert appears

=> app crashes

```
=================================================================
Main Thread Checker: UI API called on a background thread: -[UIImagePickerController init]
PID: 14963, TID: 2817330, Thread name: (none), Queue name: com.apple.mobileslideshow.accessCallbacks, QoS: 0
Backtrace:
4   BeaconExample                       0x0000000102fa6b14 -[HSAttachmentPicker showImagePicker:] + 52
5   Photos                              0x000000019d7ca910 57019CCD-D490-3268-8CEE-7FC77D5BDCA4 + 915728
6   AssetsLibraryServices               0x00000001a6af8f4c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 450380
7   AssetsLibraryServices               0x00000001a6ac2d1c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 228636
8   libdispatch.dylib                   0x0000000103135e1c _dispatch_call_block_and_release + 32
9   libdispatch.dylib                   0x000000010313727c _dispatch_client_callout + 20
10  libdispatch.dylib                   0x000000010313e90c _dispatch_lane_serial_drain + 720
11  libdispatch.dylib                   0x000000010313f4fc _dispatch_lane_invoke + 408
12  libdispatch.dylib                   0x000000010314a4dc _dispatch_workloop_worker_thread + 1344
13  libsystem_pthread.dylib             0x0000000191ccfa94 _pthread_wqthread + 280
14  libsystem_pthread.dylib             0x0000000191cd5c7c start_wqthread + 8
2019-11-27 12:11:03.626425+0100 BeaconExample[14963:2817330] [reports] Main Thread Checker: UI API called on a background thread: -[UIImagePickerController init]
PID: 14963, TID: 2817330, Thread name: (none), Queue name: com.apple.mobileslideshow.accessCallbacks, QoS: 0
Backtrace:
4   BeaconExample                       0x0000000102fa6b14 -[HSAttachmentPicker showImagePicker:] + 52
5   Photos                              0x000000019d7ca910 57019CCD-D490-3268-8CEE-7FC77D5BDCA4 + 915728
6   AssetsLibraryServices               0x00000001a6af8f4c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 450380
7   AssetsLibraryServices               0x00000001a6ac2d1c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 228636
8   libdispatch.dylib                   0x0000000103135e1c _dispatch_call_block_and_release + 32
9   libdispatch.dylib                   0x000000010313727c _dispatch_client_callout + 20
10  libdispatch.dylib                   0x000000010313e90c _dispatch_lane_serial_drain + 720
11  libdispatch.dylib                   0x000000010313f4fc _dispatch_lane_invoke + 408
12  libdispatch.dylib                   0x000000010314a4dc _dispatch_workloop_worker_thread + 1344
13  libsystem_pthread.dylib             0x0000000191ccfa94 _pthread_wqthread + 280
14  libsystem_pthread.dylib             0x0000000191cd5c7c start_wqthread + 8
=================================================================
Main Thread Checker: UI API called on a background thread: -[UINavigationController setDelegate:]
PID: 14963, TID: 2817330, Thread name: (none), Queue name: com.apple.mobileslideshow.accessCallbacks, QoS: 0
Backtrace:
4   BeaconExample                       0x0000000102fa6b28 -[HSAttachmentPicker showImagePicker:] + 72
5   Photos                              0x000000019d7ca910 57019CCD-D490-3268-8CEE-7FC77D5BDCA4 + 915728
6   AssetsLibraryServices               0x00000001a6af8f4c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 450380
7   AssetsLibraryServices               0x00000001a6ac2d1c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 228636
8   libdispatch.dylib                   0x0000000103135e1c _dispatch_call_block_and_release + 32
9   libdispatch.dylib                   0x000000010313727c _dispatch_client_callout + 20
10  libdispatch.dylib                   0x000000010313e90c _dispatch_lane_serial_drain + 720
11  libdispatch.dylib                   0x000000010313f4fc _dispatch_lane_invoke + 408
12  libdispatch.dylib                   0x000000010314a4dc _dispatch_workloop_worker_thread + 1344
13  libsystem_pthread.dylib             0x0000000191ccfa94 _pthread_wqthread + 280
14  libsystem_pthread.dylib             0x0000000191cd5c7c start_wqthread + 8
2019-11-27 12:11:13.035711+0100 BeaconExample[14963:2817330] [reports] Main Thread Checker: UI API called on a background thread: -[UINavigationController setDelegate:]
PID: 14963, TID: 2817330, Thread name: (none), Queue name: com.apple.mobileslideshow.accessCallbacks, QoS: 0
Backtrace:
4   BeaconExample                       0x0000000102fa6b28 -[HSAttachmentPicker showImagePicker:] + 72
5   Photos                              0x000000019d7ca910 57019CCD-D490-3268-8CEE-7FC77D5BDCA4 + 915728
6   AssetsLibraryServices               0x00000001a6af8f4c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 450380
7   AssetsLibraryServices               0x00000001a6ac2d1c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 228636
8   libdispatch.dylib                   0x0000000103135e1c _dispatch_call_block_and_release + 32
9   libdispatch.dylib                   0x000000010313727c _dispatch_client_callout + 20
10  libdispatch.dylib                   0x000000010313e90c _dispatch_lane_serial_drain + 720
11  libdispatch.dylib                   0x000000010313f4fc _dispatch_lane_invoke + 408
12  libdispatch.dylib                   0x000000010314a4dc _dispatch_workloop_worker_thread + 1344
13  libsystem_pthread.dylib             0x0000000191ccfa94 _pthread_wqthread + 280
14  libsystem_pthread.dylib             0x0000000191cd5c7c start_wqthread + 8
=================================================================
Main Thread Checker: UI API called on a background thread: -[UIImagePickerController setAllowsEditing:]
PID: 14963, TID: 2817330, Thread name: (none), Queue name: com.apple.mobileslideshow.accessCallbacks, QoS: 0
Backtrace:
4   BeaconExample                       0x0000000102fa6b3c -[HSAttachmentPicker showImagePicker:] + 92
5   Photos                              0x000000019d7ca910 57019CCD-D490-3268-8CEE-7FC77D5BDCA4 + 915728
6   AssetsLibraryServices               0x00000001a6af8f4c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 450380
7   AssetsLibraryServices               0x00000001a6ac2d1c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 228636
8   libdispatch.dylib                   0x0000000103135e1c _dispatch_call_block_and_release + 32
9   libdispatch.dylib                   0x000000010313727c _dispatch_client_callout + 20
10  libdispatch.dylib                   0x000000010313e90c _dispatch_lane_serial_drain + 720
11  libdispatch.dylib                   0x000000010313f4fc _dispatch_lane_invoke + 408
12  libdispatch.dylib                   0x000000010314a4dc _dispatch_workloop_worker_thread + 1344
13  libsystem_pthread.dylib             0x0000000191ccfa94 _pthread_wqthread + 280
14  libsystem_pthread.dylib             0x0000000191cd5c7c start_wqthread + 8
2019-11-27 12:11:13.133969+0100 BeaconExample[14963:2817330] [reports] Main Thread Checker: UI API called on a background thread: -[UIImagePickerController setAllowsEditing:]
PID: 14963, TID: 2817330, Thread name: (none), Queue name: com.apple.mobileslideshow.accessCallbacks, QoS: 0
Backtrace:
4   BeaconExample                       0x0000000102fa6b3c -[HSAttachmentPicker showImagePicker:] + 92
5   Photos                              0x000000019d7ca910 57019CCD-D490-3268-8CEE-7FC77D5BDCA4 + 915728
6   AssetsLibraryServices               0x00000001a6af8f4c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 450380
7   AssetsLibraryServices               0x00000001a6ac2d1c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 228636
8   libdispatch.dylib                   0x0000000103135e1c _dispatch_call_block_and_release + 32
9   libdispatch.dylib                   0x000000010313727c _dispatch_client_callout + 20
10  libdispatch.dylib                   0x000000010313e90c _dispatch_lane_serial_drain + 720
11  libdispatch.dylib                   0x000000010313f4fc _dispatch_lane_invoke + 408
12  libdispatch.dylib                   0x000000010314a4dc _dispatch_workloop_worker_thread + 1344
13  libsystem_pthread.dylib             0x0000000191ccfa94 _pthread_wqthread + 280
14  libsystem_pthread.dylib             0x0000000191cd5c7c start_wqthread + 8
=================================================================
Main Thread Checker: UI API called on a background thread: -[UIImagePickerController setMediaTypes:]
PID: 14963, TID: 2817330, Thread name: (none), Queue name: com.apple.mobileslideshow.accessCallbacks, QoS: 0
Backtrace:
4   BeaconExample                       0x0000000102fa6b88 -[HSAttachmentPicker showImagePicker:] + 168
5   Photos                              0x000000019d7ca910 57019CCD-D490-3268-8CEE-7FC77D5BDCA4 + 915728
6   AssetsLibraryServices               0x00000001a6af8f4c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 450380
7   AssetsLibraryServices               0x00000001a6ac2d1c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 228636
8   libdispatch.dylib                   0x0000000103135e1c _dispatch_call_block_and_release + 32
9   libdispatch.dylib                   0x000000010313727c _dispatch_client_callout + 20
10  libdispatch.dylib                   0x000000010313e90c _dispatch_lane_serial_drain + 720
11  libdispatch.dylib                   0x000000010313f4fc _dispatch_lane_invoke + 408
12  libdispatch.dylib                   0x000000010314a4dc _dispatch_workloop_worker_thread + 1344
13  libsystem_pthread.dylib             0x0000000191ccfa94 _pthread_wqthread + 280
14  libsystem_pthread.dylib             0x0000000191cd5c7c start_wqthread + 8
2019-11-27 12:11:13.193997+0100 BeaconExample[14963:2817330] [reports] Main Thread Checker: UI API called on a background thread: -[UIImagePickerController setMediaTypes:]
PID: 14963, TID: 2817330, Thread name: (none), Queue name: com.apple.mobileslideshow.accessCallbacks, QoS: 0
Backtrace:
4   BeaconExample                       0x0000000102fa6b88 -[HSAttachmentPicker showImagePicker:] + 168
5   Photos                              0x000000019d7ca910 57019CCD-D490-3268-8CEE-7FC77D5BDCA4 + 915728
6   AssetsLibraryServices               0x00000001a6af8f4c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 450380
7   AssetsLibraryServices               0x00000001a6ac2d1c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 228636
8   libdispatch.dylib                   0x0000000103135e1c _dispatch_call_block_and_release + 32
9   libdispatch.dylib                   0x000000010313727c _dispatch_client_callout + 20
10  libdispatch.dylib                   0x000000010313e90c _dispatch_lane_serial_drain + 720
11  libdispatch.dylib                   0x000000010313f4fc _dispatch_lane_invoke + 408
12  libdispatch.dylib                   0x000000010314a4dc _dispatch_workloop_worker_thread + 1344
13  libsystem_pthread.dylib             0x0000000191ccfa94 _pthread_wqthread + 280
14  libsystem_pthread.dylib             0x0000000191cd5c7c start_wqthread + 8
=================================================================
Main Thread Checker: UI API called on a background thread: -[UIImagePickerController setVideoQuality:]
PID: 14963, TID: 2817330, Thread name: (none), Queue name: com.apple.mobileslideshow.accessCallbacks, QoS: 0
Backtrace:
4   BeaconExample                       0x0000000102fa6bb4 -[HSAttachmentPicker showImagePicker:] + 212
5   Photos                              0x000000019d7ca910 57019CCD-D490-3268-8CEE-7FC77D5BDCA4 + 915728
6   AssetsLibraryServices               0x00000001a6af8f4c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 450380
7   AssetsLibraryServices               0x00000001a6ac2d1c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 228636
8   libdispatch.dylib                   0x0000000103135e1c _dispatch_call_block_and_release + 32
9   libdispatch.dylib                   0x000000010313727c _dispatch_client_callout + 20
10  libdispatch.dylib                   0x000000010313e90c _dispatch_lane_serial_drain + 720
11  libdispatch.dylib                   0x000000010313f4fc _dispatch_lane_invoke + 408
12  libdispatch.dylib                   0x000000010314a4dc _dispatch_workloop_worker_thread + 1344
13  libsystem_pthread.dylib             0x0000000191ccfa94 _pthread_wqthread + 280
14  libsystem_pthread.dylib             0x0000000191cd5c7c start_wqthread + 8
2019-11-27 12:11:13.251680+0100 BeaconExample[14963:2817330] [reports] Main Thread Checker: UI API called on a background thread: -[UIImagePickerController setVideoQuality:]
PID: 14963, TID: 2817330, Thread name: (none), Queue name: com.apple.mobileslideshow.accessCallbacks, QoS: 0
Backtrace:
4   BeaconExample                       0x0000000102fa6bb4 -[HSAttachmentPicker showImagePicker:] + 212
5   Photos                              0x000000019d7ca910 57019CCD-D490-3268-8CEE-7FC77D5BDCA4 + 915728
6   AssetsLibraryServices               0x00000001a6af8f4c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 450380
7   AssetsLibraryServices               0x00000001a6ac2d1c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 228636
8   libdispatch.dylib                   0x0000000103135e1c _dispatch_call_block_and_release + 32
9   libdispatch.dylib                   0x000000010313727c _dispatch_client_callout + 20
10  libdispatch.dylib                   0x000000010313e90c _dispatch_lane_serial_drain + 720
11  libdispatch.dylib                   0x000000010313f4fc _dispatch_lane_invoke + 408
12  libdispatch.dylib                   0x000000010314a4dc _dispatch_workloop_worker_thread + 1344
13  libsystem_pthread.dylib             0x0000000191ccfa94 _pthread_wqthread + 280
14  libsystem_pthread.dylib             0x0000000191cd5c7c start_wqthread + 8
=================================================================
Main Thread Checker: UI API called on a background thread: -[UIImagePickerController setSourceType:]
PID: 14963, TID: 2817330, Thread name: (none), Queue name: com.apple.mobileslideshow.accessCallbacks, QoS: 0
Backtrace:
4   BeaconExample                       0x0000000102fa6bc8 -[HSAttachmentPicker showImagePicker:] + 232
5   Photos                              0x000000019d7ca910 57019CCD-D490-3268-8CEE-7FC77D5BDCA4 + 915728
6   AssetsLibraryServices               0x00000001a6af8f4c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 450380
7   AssetsLibraryServices               0x00000001a6ac2d1c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 228636
8   libdispatch.dylib                   0x0000000103135e1c _dispatch_call_block_and_release + 32
9   libdispatch.dylib                   0x000000010313727c _dispatch_client_callout + 20
10  libdispatch.dylib                   0x000000010313e90c _dispatch_lane_serial_drain + 720
11  libdispatch.dylib                   0x000000010313f4fc _dispatch_lane_invoke + 408
12  libdispatch.dylib                   0x000000010314a4dc _dispatch_workloop_worker_thread + 1344
13  libsystem_pthread.dylib             0x0000000191ccfa94 _pthread_wqthread + 280
14  libsystem_pthread.dylib             0x0000000191cd5c7c start_wqthread + 8
2019-11-27 12:11:13.302273+0100 BeaconExample[14963:2817330] [reports] Main Thread Checker: UI API called on a background thread: -[UIImagePickerController setSourceType:]
PID: 14963, TID: 2817330, Thread name: (none), Queue name: com.apple.mobileslideshow.accessCallbacks, QoS: 0
Backtrace:
4   BeaconExample                       0x0000000102fa6bc8 -[HSAttachmentPicker showImagePicker:] + 232
5   Photos                              0x000000019d7ca910 57019CCD-D490-3268-8CEE-7FC77D5BDCA4 + 915728
6   AssetsLibraryServices               0x00000001a6af8f4c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 450380
7   AssetsLibraryServices               0x00000001a6ac2d1c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 228636
8   libdispatch.dylib                   0x0000000103135e1c _dispatch_call_block_and_release + 32
9   libdispatch.dylib                   0x000000010313727c _dispatch_client_callout + 20
10  libdispatch.dylib                   0x000000010313e90c _dispatch_lane_serial_drain + 720
11  libdispatch.dylib                   0x000000010313f4fc _dispatch_lane_invoke + 408
12  libdispatch.dylib                   0x000000010314a4dc _dispatch_workloop_worker_thread + 1344
13  libsystem_pthread.dylib             0x0000000191ccfa94 _pthread_wqthread + 280
14  libsystem_pthread.dylib             0x0000000191cd5c7c start_wqthread + 8
=================================================================
Main Thread Checker: UI API called on a background thread: -[UIViewController popoverPresentationController]
PID: 14963, TID: 2817330, Thread name: (none), Queue name: com.apple.mobileslideshow.accessCallbacks, QoS: 0
Backtrace:
4   BeaconExample                       0x0000000102fa152c -[HSBeaconMessageEntryContainerController attachmentPickerMenu:showController:completion:] + 68
5   BeaconExample                       0x0000000102fa6c00 -[HSAttachmentPicker showImagePicker:] + 288
6   Photos                              0x000000019d7ca910 57019CCD-D490-3268-8CEE-7FC77D5BDCA4 + 915728
7   AssetsLibraryServices               0x00000001a6af8f4c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 450380
8   AssetsLibraryServices               0x00000001a6ac2d1c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 228636
9   libdispatch.dylib                   0x0000000103135e1c _dispatch_call_block_and_release + 32
10  libdispatch.dylib                   0x000000010313727c _dispatch_client_callout + 20
11  libdispatch.dylib                   0x000000010313e90c _dispatch_lane_serial_drain + 720
12  libdispatch.dylib                   0x000000010313f4fc _dispatch_lane_invoke + 408
13  libdispatch.dylib                   0x000000010314a4dc _dispatch_workloop_worker_thread + 1344
14  libsystem_pthread.dylib             0x0000000191ccfa94 _pthread_wqthread + 280
15  libsystem_pthread.dylib             0x0000000191cd5c7c start_wqthread + 8
2019-11-27 12:11:13.400008+0100 BeaconExample[14963:2817330] [reports] Main Thread Checker: UI API called on a background thread: -[UIViewController popoverPresentationController]
PID: 14963, TID: 2817330, Thread name: (none), Queue name: com.apple.mobileslideshow.accessCallbacks, QoS: 0
Backtrace:
4   BeaconExample                       0x0000000102fa152c -[HSBeaconMessageEntryContainerController attachmentPickerMenu:showController:completion:] + 68
5   BeaconExample                       0x0000000102fa6c00 -[HSAttachmentPicker showImagePicker:] + 288
6   Photos                              0x000000019d7ca910 57019CCD-D490-3268-8CEE-7FC77D5BDCA4 + 915728
7   AssetsLibraryServices               0x00000001a6af8f4c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 450380
8   AssetsLibraryServices               0x00000001a6ac2d1c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 228636
9   libdispatch.dylib                   0x0000000103135e1c _dispatch_call_block_and_release + 32
10  libdispatch.dylib                   0x000000010313727c _dispatch_client_callout + 20
11  libdispatch.dylib                   0x000000010313e90c _dispatch_lane_serial_drain + 720
12  libdispatch.dylib                   0x000000010313f4fc _dispatch_lane_invoke + 408
13  libdispatch.dylib                   0x000000010314a4dc _dispatch_workloop_worker_thread + 1344
14  libsystem_pthread.dylib             0x0000000191ccfa94 _pthread_wqthread + 280
15  libsystem_pthread.dylib             0x0000000191cd5c7c start_wqthread + 8
=================================================================
Main Thread Checker: UI API called on a background thread: -[UIViewController presentViewController:animated:completion:]
PID: 14963, TID: 2817330, Thread name: (none), Queue name: com.apple.mobileslideshow.accessCallbacks, QoS: 0
Backtrace:
4   BeaconExample                       0x0000000102fa15a4 -[HSBeaconMessageEntryContainerController attachmentPickerMenu:showController:completion:] + 188
5   BeaconExample                       0x0000000102fa6c00 -[HSAttachmentPicker showImagePicker:] + 288
6   Photos                              0x000000019d7ca910 57019CCD-D490-3268-8CEE-7FC77D5BDCA4 + 915728
7   AssetsLibraryServices               0x00000001a6af8f4c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 450380
8   AssetsLibraryServices               0x00000001a6ac2d1c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 228636
9   libdispatch.dylib                   0x0000000103135e1c _dispatch_call_block_and_release + 32
10  libdispatch.dylib                   0x000000010313727c _dispatch_client_callout + 20
11  libdispatch.dylib                   0x000000010313e90c _dispatch_lane_serial_drain + 720
12  libdispatch.dylib                   0x000000010313f4fc _dispatch_lane_invoke + 408
13  libdispatch.dylib                   0x000000010314a4dc _dispatch_workloop_worker_thread + 1344
14  libsystem_pthread.dylib             0x0000000191ccfa94 _pthread_wqthread + 280
15  libsystem_pthread.dylib             0x0000000191cd5c7c start_wqthread + 8
2019-11-27 12:11:13.460845+0100 BeaconExample[14963:2817330] [reports] Main Thread Checker: UI API called on a background thread: -[UIViewController presentViewController:animated:completion:]
PID: 14963, TID: 2817330, Thread name: (none), Queue name: com.apple.mobileslideshow.accessCallbacks, QoS: 0
Backtrace:
4   BeaconExample                       0x0000000102fa15a4 -[HSBeaconMessageEntryContainerController attachmentPickerMenu:showController:completion:] + 188
5   BeaconExample                       0x0000000102fa6c00 -[HSAttachmentPicker showImagePicker:] + 288
6   Photos                              0x000000019d7ca910 57019CCD-D490-3268-8CEE-7FC77D5BDCA4 + 915728
7   AssetsLibraryServices               0x00000001a6af8f4c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 450380
8   AssetsLibraryServices               0x00000001a6ac2d1c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 228636
9   libdispatch.dylib                   0x0000000103135e1c _dispatch_call_block_and_release + 32
10  libdispatch.dylib                   0x000000010313727c _dispatch_client_callout + 20
11  libdispatch.dylib                   0x000000010313e90c _dispatch_lane_serial_drain + 720
12  libdispatch.dylib                   0x000000010313f4fc _dispatch_lane_invoke + 408
13  libdispatch.dylib                   0x000000010314a4dc _dispatch_workloop_worker_thread + 1344
14  libsystem_pthread.dylib             0x0000000191ccfa94 _pthread_wqthread + 280
15  libsystem_pthread.dylib             0x0000000191cd5c7c start_wqthread + 8
2019-11-27 12:11:13.544279+0100 BeaconExample[14963:2817330] [Assert] Cannot be called with asCopy = NO on non-main thread.
2019-11-27 12:11:13.544688+0100 BeaconExample[14963:2817330] [Assert] Cannot be called with asCopy = NO on non-main thread.
2019-11-27 12:11:13.552200+0100 BeaconExample[14963:2817330] [Assert] Cannot be called with asCopy = NO on non-main thread.
2019-11-27 12:11:13.552530+0100 BeaconExample[14963:2817330] [Assert] Cannot be called with asCopy = NO on non-main thread.
2019-11-27 12:11:13.553376+0100 BeaconExample[14963:2817330] [Assert] Cannot be called with asCopy = NO on non-main thread.
2019-11-27 12:11:13.553587+0100 BeaconExample[14963:2817330] [Assert] Cannot be called with asCopy = NO on non-main thread.
2019-11-27 12:11:13.555451+0100 BeaconExample[14963:2817330] [Assert] Cannot be called with asCopy = NO on non-main thread.
2019-11-27 12:11:13.555678+0100 BeaconExample[14963:2817330] [Assert] Cannot be called with asCopy = NO on non-main thread.
2019-11-27 12:11:13.555883+0100 BeaconExample[14963:2817330] [Assert] Cannot be called with asCopy = NO on non-main thread.
2019-11-27 12:11:13.556104+0100 BeaconExample[14963:2817330] [Assert] Cannot be called with asCopy = NO on non-main thread.
2019-11-27 12:11:13.556291+0100 BeaconExample[14963:2817330] [Assert] Cannot be called with asCopy = NO on non-main thread.
2019-11-27 12:11:13.556402+0100 BeaconExample[14963:2817330] [Assert] Cannot be called with asCopy = NO on non-main thread.
2019-11-27 12:11:13.567608+0100 BeaconExample[14963:2817330] [Animation] +[UIView setAnimationsEnabled:] being called from a background thread. Performing any operation from a background thread on UIView or a subclass is not supported and may result in unexpected and insidious behavior. trace=(
	0   UIKitCore                           0x000000019654e11c A5A8BFF2-89EC-372C-B97C-D0431C1F88E0 + 15626524
	1   UIKitCore                           0x000000019654e1b4 A5A8BFF2-89EC-372C-B97C-D0431C1F88E0 + 15626676
	2   libdispatch.dylib                   0x000000010313727c _dispatch_client_callout + 20
	3   libdispatch.dylib                   0x0000000103138c88 _dispatch_once_callout + 132
	4   UIKitCore                           0x000000019654e0a0 A5A8BFF2-89EC-372C-B97C-D0431C1F88E0 + 15626400
	5   UIKitCore                           0x000000019654e220 A5A8BFF2-89EC-372C-B97C-D0431C1F88E0 + 15626784
	6   UIKitCore                           0x0000000195f0bee8 A5A8BFF2-89EC-372C-B97C-D0431C1F88E0 + 9064168
	7   UIKitCore                           0x00000001962e3a64 A5A8BFF2-89EC-372C-B97C-D0431C1F88E0 + 13093476
	8   UIKitCore                           0x00000001960f6c34 A5A8BFF2-89EC-372C-B97C-D0431C1F88E0 + 11074612
	9   UIKitCore                           0x0000000195a64a74 A5A8BFF2-89EC-372C-B97C-D0431C1F88E0 + 4184692
	10  UIKitCore                           0x0000000195a649ec A5A8BFF2-89EC-372C-B97C-D0431C1F88E0 + 4184556
	11  UIKitCore                           0x0000000195a84c20 A5A8BFF2-89EC-372C-B97C-D0431C1F88E0 + 4316192
	12  UIKitCore                           0x0000000195a75654 A5A8BFF2-89EC-372C-B97C-D0431C1F88E0 + 4253268
	13  UIKitCore                           0x0000000195a77024 A5A8BFF2-89EC-372C-B97C-D0431C1F88E0 + 4259876
	14  UIKitCore                           0x0000000195a795d4 A5A8BFF2-89EC-372C-B97C-D0431C1F88E0 + 4269524
	15  UIKitCore                           0x0000000195a79b38 A5A8BFF2-89EC-372C-B97C-D0431C1F88E0 + 4270904
	16  UIKitCore                           0x0000000195a79520 A5A8BFF2-89EC-372C-B97C-D0431C1F88E0 + 4269344
	17  UIKitCore                           0x0000000195a797b4 A5A8BFF2-89EC-372C-B97C-D0431C1F88E0 + 4270004
	18  BeaconExample                       0x0000000102fa15a4 -[HSBeaconMessageEntryContainerController attachmentPickerMenu:showController:completion:] + 188
	19  BeaconExample                       0x0000000102fa6c00 -[HSAttachmentPicker showImagePicker:] + 288
	20  Photos                              0x000000019d7ca910 57019CCD-D490-3268-8CEE-7FC77D5BDCA4 + 915728
	21  AssetsLibraryServices               0x00000001a6af8f4c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 450380
	22  AssetsLibraryServices               0x00000001a6ac2d1c A93F27C0-B64D-30BD-BDCD-D2FB968F6C6F + 228636
	23  libdispatch.dylib                   0x0000000103135e1c _dispatch_call_block_and_release + 32
	24  libdispatch.dylib                   0x000000010313727c _dispatch_client_callout + 20
	25  libdispatch.dylib                   0x000000010313e90c _dispatch_lane_serial_drain + 720
	26  libdispatch.dylib                   0x000000010313f4fc _dispatch_lane_invoke + 408
	27  libdispatch.dylib                   0x000000010314a4dc _dispatch_workloop_worker_thread + 1344
	28  libsystem_pthread.dylib             0x0000000191ccfa94 _pthread_wqthread + 280
	29  libsystem_pthread.dylib             0x0000000191cd5c7c start_wqthread + 8
)
2019-11-27 12:11:13.568772+0100 BeaconExample[14963:2817330] [Assert] Cannot be called with asCopy = NO on non-main thread.
2019-11-27 12:11:13.569006+0100 BeaconExample[14963:2817330] [Assert] Cannot be called with asCopy = NO on non-main thread.
2019-11-27 12:11:13.569813+0100 BeaconExample[14963:2817330] *** Assertion failure in -[FBSSerialQueue assertOnQueue], /BuildRoot/Library/Caches/com.apple.xbs/Sources/FrontBoardServices/FrontBoard-626.2/FrontBoardServices/FBSSerialQueue.m:98
2019-11-27 12:11:13.570943+0100 BeaconExample[14963:2817330] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'threading violation: expected the main thread'
*** First throw call stack:
(0x191fc3ab0 0x191cdd028 0x191ec06ac 0x192301720 0x1971da89c 0x19718c510 0x19718c79c 0x19718c694 0x195c1f370 0x19584ca00 0x195a85004 0x195a64ac0 0x195a649ec 0x195a84c20 0x195a75654 0x195a77024 0x195a795d4 0x195a79b38 0x195a79520 0x195a797b4 0x102fa15a4 0x102fa6c00 0x19d7ca910 0x1a6af8f4c 0x1a6ac2d1c 0x103135e1c 0x10313727c 0x10313e90c 0x10313f4fc 0x10314a4dc 0x191ccfa94 0x191cd5c7c)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

**Comment by samsymons** [comment]
@imanoupetit Thanks for the steps! We'll dig into this and fix this for a future Beacon release.

Internal Jira reference: https://helpscout.atlassian.net/browse/IOSSDK-47

**Comment by samsymons** [comment]
Thanks again for the report here! This issue should be fixed in Beacon 1.0.3. ♥️
